### DISCUSSION TITLE
Pagination by years

### DISCUSSION THREAD

**Discussion by oim37** [wintercms/wn-blog-plugin#57] [Q&A] [answered]
Hello,

I have pagination by pages number:

```
{% for post in posts %}
{% for category in post.categories %}
  {% if category.isLeaf() %}
    [{{ category.name }}]
  {% endif %}
{% endfor %}
{{post.published_at|date('d.m.Y')}}<br/>
<b>{{post.title}}</b><br/>
{{post.excerpt}}<br/>
<a href="{{post.url}}"><div class="item-news-block__link link-line"><u>{{ 'nav.more'|_ }}</u></div></a><br/><br/>
{% endfor %}

{% if posts.lastPage > 1 %}
 <ul class="pagination">
 {% if posts.currentPage > 1 %}
	<li><a href="{{ this.page.baseFileName | page({ (blogPosts.pageParam): (posts.currentPage-1) })}}"><u>{{ 'nav.prev'|_ }}</u></a>&nbsp;&nbsp;&nbsp;</li>
	{% endif %}
	
	{% for page in 1..posts.lastPage %}
	<li class="{{ ( posts.currentPage == page ? 'active-page' : null ) }}">
	<a href="{{ this.page.baseFineName|page({ (pageParam): page }) }}"><u>{{ page }}</u>&nbsp;</a>
		</li>
		{% endfor %}
		
	{% if posts.lastPage > posts.currentPage %}
	<li><a href="{{ this.page.baseFileName | page({ (blogPosts.pageParam): (posts.currentPage+1) })  }}">&nbsp;&nbsp;<u>{{ 'nav.next'|_ }}</u></a></li>
	{% endif %}
	</ul>
	{% endif %}
```

Logically, I decided that this was not convenient from the user’s point of view; it would be much more interesting to do the year instead of page numbers.

Now I have list of 6 entries per page and page numbers at the bottom.

<1 2 3 4 5>

How to make list records sorted and displayed by each year?

< 2023 2022 2021 2020 2019 >

Thank you in advance.

**Comment by bennothommo** [comment, marked as answer]
My thinking is that you will probably have to do a custom component to make this happen.

You could perhaps create a plugin and component that extends the `\Winter\Blog\Components\Posts` component, and change up the `listPosts()` method to instead group the collection of posts by year (using the `groupBy()` method and function that returns the year of each post).

You would also have to change up the component partials to accept this new format.

**Comment by oim37** [comment]
Ok, thank you.
